File: frontend/src/lib/marketplacePricing.ts
export interface EbayPricingInput {
  vintedPrice: number;
}

export interface EbayPricingResult {
  beforeRounding: number;
  roundingStep: 0.5 | 1;
  ebayPrice: number;
}

export function calculateEbayPrice(input: EbayPricingInput): EbayPricingResult {
  if (!Number.isFinite(input.vintedPrice) || input.vintedPrice < 0) {
    throw new Error('Le prix doit être un nombre positif ou nul.');
  }

  const beforeRounding = (input.vintedPrice + 0.35) / 0.91;
  const roundingStep: 0.5 | 1 = beforeRounding < 5 ? 0.5 : 1;
  const ebayPrice = Math.ceil((beforeRounding - Number.EPSILON) / roundingStep) * roundingStep;

  return {
    beforeRounding: Math.round(beforeRounding * 100) / 100,
    roundingStep,
    ebayPrice: Math.round(ebayPrice * 100) / 100,
  };
}

File: backend/services/marketplace_pricing.py
import math


def calculate_ebay_price(vinted_price: float) -> float:
    if not math.isfinite(vinted_price) or vinted_price < 0:
        raise ValueError("Le prix Vinted doit être positif ou nul.")
    before_rounding = (vinted_price + 0.35) / 0.91
    step = 0.5 if before_rounding < 5 else 1
    return round(math.ceil((before_rounding - 1e-12) / step) * step, 2)
